File: api/WebShop.Domain/Models/Company.cs
using WebShop.Domain.Enums;

namespace WebShop.Domain.Models
{
    public class Company : BaseType
    {
        public string ExternalId { get; set; }
        public string Surname { get; set; }
        public string Address { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string City { get; set; }
        public string PostCode { get; set; }
        public DeliveryType DeliveryType { get; set; }
    }
}

File: spa/webshop/src/app/models/company.ts
import { Item } from './item';

export interface Company {
  id: string;
  name: string;
  externalId: string;
  surname: string;
  address: string;
  phone: string;
  email: string;
  city: string;
  cityCode: number;
  street: string;
  deliveryType: DeliveryType;
  items: Item[];
}

export interface DeliveryType {
  delivery: string;
  pickUp: string;
}
